File: solutionFiles/ValidParenthese.py
import json
import sys

def is_valid_parentheses(s):
    stack = []
    mapping = {')': '(', '}': '{', ']': '['}

    for char in s:
        if char in mapping.values():
            stack.append(char)
        elif char in mapping:
            if stack and stack[-1] == mapping[char]:
                stack.pop()
            else:
                return False
        else:
            return False

    return not stack

def main():
    # Read the JSON string from program argument
    input_json = sys.argv[1]
    input_data = json.loads(input_json)

    s = input_data['s']
    result = is_valid_parentheses(s)

    # Output the result as JSON string
    output_json = json.dumps({"result": result})
    print(output_json)

if __name__ == "__main__":
    main()

File: solutionFiles/ValidParenthese.js
const process = require('process');

function isValidParentheses(s) {
    const stack = [];
    const mapping = { ')': '(', '}': '{', ']': '[' };

    for (let char of s) {
        if (Object.values(mapping).includes(char)) {
            stack.push(char);
        } else if (Object.keys(mapping).includes(char)) {
            if (stack.length && stack[stack.length - 1] === mapping[char]) {
                stack.pop();
            } else {
                return false;
            }
        } else {
            return false;
        }
    }

    return stack.length === 0;
}

function main() {
    // Read the JSON string from program argument
    const inputJson = process.argv[2];
    const inputData = JSON.parse(inputJson);

    const s = inputData.s;
    const result = isValidParentheses(s);

    // Output the result as JSON string
    const outputJson = JSON.stringify({ result: result });
    console.log(outputJson);
}

if (require.main === module) {
    main();
}
